src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py

```diff
@@ -227,3 +227,25 @@ def create_base_dir(self, mode):
             self.fs.mkdirs(self.base_path, mode)
         except cephfs.Error as e:
             raise VolumeException(-e.args[0], e.args[1])
+
+    def info (self):
+        subvolpath = self.metadata_mgr.get_global_option('path')
+        etype = self.metadata_mgr.get_global_option(MetadataManager.GLOBAL_META_KEY_TYPE)
+        st = self.fs.statx(subvolpath, cephfs.CEPH_STATX_BTIME |
+                                            cephfs.CEPH_STATX_SIZE,
+                                            cephfs.AT_SYMLINK_NOFOLLOW)
+        usedbytes = st["size"]
+        try:
+            nsize = int(self.fs.getxattr(subvolpath, 'ceph.quota.max_bytes').decode('utf-8'))
+        except cephfs.NoData:
+            nsize = 0
+
+        try:
+            data_pool = self.fs.getxattr(subvolpath, 'ceph.dir.layout.pool').decode('utf-8')
+        except cephfs.Error as e:
+            raise VolumeException(-e.args[0], e.args[1])
+
+        return {'path': subvolpath, 'type': etype,
+            'bytes_quota': "no-limit" if nsize == 0 else nsize, 'bytes_used': int(usedbytes),
+            'bytes_pcent': "no-limit" if nsize == 0 else '{0:.2f}'.format((float(usedbytes) / nsize) * 100.0),
+            'data_pool': data_pool, 'created_at': str(st["btime"])}
```

Comment:
Use json.dumps

Reply:
Just return the values from here and use the appropriate json.dumps format in volume.py.

We use this format for the other commands:
```
$ ./bin/ceph fs subvolume resize myfs mysubvol 100 --group_name mygroup
[
    {
        "bytes_used": 1048576
    },
    {
        "bytes_quota": 100
    },
    {
        "bytes_pcent": "1048576.00"
    }
]

```

Reply:
All the attributes belong to a single subvolume. So I thought to keep it in a single dict made more sense. If we later come up with cli to get all the subvolumes info, then list of dicts makes sense. Isn't it?

Reply:
Agree with @kotreshhr

Reply:
Checking the design of the previous modules, there are no dictionary construction in the subvolume_*.py. They are in volume.py instead.

Reply:
Most of the functions returns at most two values, so returning directly makes more sense. In this case, we are returning more than 10 values. Returning them individually makes it cumbersome to handle at the caller. 

I saw _get_clone_source is returning dict. So returning dictionary should be fine right?

Reply:
makes sense to return a dictionary -- the command facing layer can convert it to json before returning.